### ISSUE TITLE
Crash when trying to edit item on Keys options page

### ISSUE THREAD

**Issue by clsid2** [clsid2/mpc-hc#251]
@adipose
I have not been able to reproduce this myself. But found a report from a user that the player hangs and then closes (without error dialog) after clicking on an entry in the Key column in attempt to set a value.

Happened on Windows 7, but not on Windows 10. Version 1.9.0.8 has the problem, 1.9.0 is ok.

Most likely related to this commit:
https://github.com/clsid2/mpc-hc/commit/0f17155cc4b42e9270359d72cea21fd6ff0ebcad

And probably unrelated to this issue, but do we correctly handle the case where CreateFontIndirect returns NULL? I think we might need a fallback font in case for example "Consolas" is unavailable?

**Comment by adipose** [comment]
No luck reproducing this, but I did add code to check the return value when creating fonts.

https://github.com/clsid2/mpc-hc/pull/256

By the way, in that commit are some rather boring changes to the dialog for keys:

![image](https://user-images.githubusercontent.com/3324395/73587532-017ab180-4472-11ea-8c0d-84d25ca4985f.png)

I couldn't find any reason it should cause an issue--most of the dialogs have those settings, but this one didn't before.

**Comment by adipose** [comment]
Other changes to that dialog include the table filter.  That patch has a fix to kill the timer, which shouldn't hurt anything (fixes a bug).

**Comment by adipose** [comment]
@clsid2 , any chance of talking to the user with this issue?

I have a couple ideas.  But it would be nice to reproduce it.

**Comment by adipose** [comment]
@clsid2 

See https://github.com/clsid2/mpc-hc/pull/261

The dialog options got changed for no good reason that I can see, and didn't match the other PPAGE dialogs.  After reviewing the options that were set, it's possible some behaviors changed, though I couldn't find anything that seemed to matter.  It may have been set automatically by the VS editor, or maybe it got copied from a non-options dialog by mistake.

The revert seems to cause no issues, so in case that might have caused this crash, I made a PR.

**Comment by adipose** [comment]
@clsid2 

See https://github.com/clsid2/mpc-hc/pull/263

Another fix for an issue I don't think happens, but is the only thing I can think of that might cause a crash for someone.  In practice, setting a null font didn't crash it for me, but it's worth avoiding it just in case.

**Comment by zme-ul** [comment]
I'm on 1.9.1.11 nightly 
------------------------

Build information:
    Version:            1.9.1.11 (d2d742e61)
    Compiler:           MSVC v19.16.27035
    Build date:         Feb  5 2020

LAV Filters:
    LAV Splitter:       0.74.1.31
    LAV Video:          0.74.1.31
    LAV Audio:          0.74.1.31
    FFmpeg compiler:    MinGW-w64 GCC 9.2.0

Operating system:
    Name:               Windows NT 10.0 (build 18363)
    Version:            10.0 (64-bit)

Hardware:
    CPU:                Intel(R) Core(TM) i5-9600K CPU @ 3.70GHz
    GPU:                NVIDIA GeForce GTX 1070 (driver version: 26.21.14.4219)

------------------------

if I try to remap fullscreen key from ALT+Enter, MPC-HC crashes completely

**Comment by adipose** [comment]
Can you export your settings and upload them?

**Comment by zme-ul** [comment]
[mpc-hc-settings.zip](https://github.com/clsid2/mpc-hc/files/4177905/mpc-hc-settings.zip)

**Comment by adipose** [comment]
Thanks--I couldn't reproduce it with your settings.  I also am on Windows 10 18363.

Can you make a video possibly?  Or at least tell me the exact sequence of key strokes/mouse clicks.

**Comment by zme-ul** [comment]
https://youtu.be/EsxN2_6_XzE

**Comment by adipose** [comment]
Thank you, that is really weird.  I cannot repro it, here.

If you are interested, I could set up a teamviewer session to try some other things.  I am out of ideas to repro it on my end.

**Comment by zme-ul** [comment]
neah, it's 1AM and I'm heading to work in the morning
I can use it even without remapping keys

**Comment by adipose** [comment]
No worries!

Curious, why is the window so big?  By default it's much smaller, and even importing your settings mine was small:

![image](https://user-images.githubusercontent.com/3324395/74112180-475d0700-4b4f-11ea-873e-77e278261934.png)

**Comment by adipose** [comment]
Also if you could leave your display res, dpi zoom, and font zoom settings for me to test.

**Comment by clsid2** [comment]
It is weird that the player closes without crash dialog. Perhaps a third party application is injecting code into the MPC-HC process.

**Comment by adipose** [comment]
@clsid2 , agreed. But somehow, something I changed has broken it, apparently.  So hopefully, it's fixable.

**Comment by adipose** [comment]
@zme-ul 

Here is a special test build : https://mega.nz/#!sAIhEQpS!ASbCd20bqQHOo63wFLYSQj6fsykz33gLF_8635rCU9o

Please see if this still crashes (it may look a bit funny on the button as I disabled rendering it to see if the cause is there).

I think I saw you actually have 32-bit, if so, here is 32-bit build:

https://mega.nz/#!UUBzEQQD!eJLsV89AxllxJA2pRK-RB2SptNTnqSrZ71RHeK9oS38

**Comment by zme-ul** [comment]
yep, this one works
yes, it was x64

**Comment by adipose** [comment]
Thanks for testing.  Can you try this build?

https://mega.nz/#!0ch1hZwC!Y7elzH4q-6ENu3AD31PIjROvKCXt1ropXpkWA3y5OrA

Again, it will behave a bit strange, but just check if it crashes or not.

Thank you!

**Comment by adipose** [comment]
https://mega.nz/#!IYBQTaTb!-h0tnQ-YefxqW4kAVT_xt2BrEBhXsoao95DG_og-irQ

Here is another test build.  This one I am guessing fixes the issue w/o side effects.

What I noticed is the dark theme goes through a different code path when drawing the button, where the font is not set to the parent window (the key edit field) font.  Since the parent window now had DS_FIXEDSYS set, this meant it would try to use the fixedsys font instead of some other font (maybe windows default font?).

Now, I set the font on the button to match the edit (as the classic theme did).  I am hoping this fixes your issue.  But it does suggest you have a problem with the fixedsys font on your machine--no idea why that would be.
